added subscription environment variable for SYNC-PERIOD-MIN

redhatHameed opened the pull request:
The operator does not allow users to change environment variables. All changes get reverted. the possible solution is to add into subscription config: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/subscription-config.md#subscription-config

add filed SyncPeriod into DBaaS-platform CR to make it configurable. 

## Description
<!-- Please include a summary of the change and link to the related Jira issue. Please add any additional motivation and context as needed. Screenshots are also welcome -->

## Verification Steps
<!--
Add the steps required to check this change. Following an example.

1. Go to `XX >> YY >> SS`
2. Create a new item `N` with the info `X`
3. Try to edit this item 
4. Check if in the left menu the feature X is not so long present.
-->

## Type of change
<!-- Please delete options that are not relevant. -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)
<!-- Please strikethrough options not relevant using two tildes ~~Text~~. Do not delete non relevant options -->
- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

tchughesiv commented on api/v1alpha1/dbaasplatform_types.go:
```diff
@@ -41,7 +41,7 @@ const (
 // DBaaSPlatformSpec defines the desired state of DBaaSPlatform
 type DBaaSPlatformSpec struct {
-	Name string `json:"name,omitempty"`
+	SyncPeriod string `json:"syncPeriod,omitempty"`
```
would it be a good idea to make this an integer instead? then convert to string in code?

redhatHameed replied:
make sense, added validation as well. updated

tchughesiv commented on config/crd/bases/dbaas.redhat.com_dbaasplatforms.yaml:
```diff
@@ -36,8 +36,10 @@ spec:
-              name:
-                type: string
+              syncPeriod:
+                maximum: 180
```
any reason 180 is the max here? just curious

jianrongzhang89 replied:
@redhatHameed you can set default: 180 instead of maximum.

redhatHameed replied:
@jianrongzhang89 Thanks default already set by provider operator.

@tchughesiv added 24h max value

tchughesiv reviewed (changes requested):
can we add a couple unit tests that verify no ENV is set when `cr.Spec.SyncPeriod` is empty? and another that verifies one IS set? also, what happens if an empty `SYNC-PERIOD-MIN` var is set? does that mean no refresh?

tchughesiv commented on api/v1alpha1/dbaasplatform_types.go:
```diff
@@ -41,7 +41,9 @@ const (
-	Name string `json:"name,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=180
+	SyncPeriod int `json:"syncPeriod,omitempty"`
```
can we add a description here?

redhatHameed replied:
added

redhatHameed commented:
> can we add a couple unit tests that verify no ENV is set when `cr.Spec.SyncPeriod` is empty? and another that verifies one IS set? also, what happens if an empty `SYNC-PERIOD-MIN` var is set? does that mean no refresh?

added unit test.

added if check if empty/nil cr.Spec.SyncPeriod don't set environment variable

tchughesiv reviewed (approved):
Lgtm

tchughesiv commented on controllers/dbaasplatform_controller_test.go:
```diff
@@ -0,0 +1,48 @@
+			},
+		}
+		BeforeEach(assertResourceCreation(cr))
+		AfterEach(assertResourceDeletion(cr))
```
Sorry, Can we check the resulting subscription here? See if env var is set or not

redhatHameed replied:
This required a complete platform installations unit test as currently not added. I have given try but there are some other issues faced, also we have dependencies on platform installation on openshift real environment and installation never completed see logs:
https://gist.github.com/redhatHameed/2ca684b6488b77193ab77b0591c933fb

tchughesiv reviewed (approved):
lgtm
